File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        mx, ans, l = float('-inf'), 0, 0
        q = collections.deque()
        q.append(root)
        
        while q:
            l +=1
            ls = 0
            for _ in range(len(q)):
                n = q.popleft()
                ls += n.val
                if n.left:
                    q.append(n.left)
                if n.right:
                    q.append(n.right)
            if mx < ls:
                mx, ans = ls, l
        return ans
            
                
                    
                    
                
            
            
        

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        int ans = 0;
        int level = 0;
        long max = Integer.MIN_VALUE;
        queue.add(root);
        while (!queue.isEmpty()) {
            level++;
            long sum = 0;
            int l = queue.size();
            for(int i = 0; i<l; i++) {
                TreeNode next = queue.poll();
                sum += next.val;
                if(next.left != null) {
                    queue.add(next.left);
                }
                if(next.right != null) {
                    queue.add(next.right);
                }
            }
            if(sum > max) {
                max = sum;
                ans = level;
            }
        }
        return ans;
    }

}